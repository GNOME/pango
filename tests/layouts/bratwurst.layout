{
  "context" : {
    "font" : "serif 12",
    "language" : "en-us",
    "base-gravity" : "south",
    "gravity-hint" : "natural",
    "base-dir" : "weak-ltr",
    "round-glyph-positions" : true,
    "transform" : [
      1,
      0,
      0,
      1,
      0,
      0
    ]
  },
  "comment" : "A case with a single word made up of two items,\nexpected to produce a single overlong line.",
  "text" : "Bratwurst",
  "attributes" : [
    {
      "start" : 4,
      "type" : "style",
      "value" : "italic"
    }
  ],
  "font" : "Cantarell Bold 32",
  "width" : 87040,
  "log-attrs" : [
    {
      "char-break" : true,
      "cursor-position" : true,
      "word-start" : true,
      "sentence-boundary" : true,
      "sentence-start" : true,
      "backspace-deletes-character" : true,
      "word-boundary" : true
    },
    {
      "char-break" : true,
      "cursor-position" : true,
      "break-inserts-hyphen" : true
    },
    {
      "char-break" : true,
      "cursor-position" : true,
      "break-inserts-hyphen" : true
    },
    {
      "char-break" : true,
      "cursor-position" : true,
      "break-inserts-hyphen" : true
    },
    {
      "char-break" : true,
      "cursor-position" : true,
      "break-inserts-hyphen" : true
    },
    {
      "char-break" : true,
      "cursor-position" : true,
      "break-inserts-hyphen" : true
    },
    {
      "char-break" : true,
      "cursor-position" : true,
      "break-inserts-hyphen" : true
    },
    {
      "char-break" : true,
      "cursor-position" : true,
      "break-inserts-hyphen" : true
    },
    {
      "char-break" : true,
      "cursor-position" : true,
      "break-inserts-hyphen" : true
    },
    {
      "line-break" : true,
      "mandatory-break" : true,
      "char-break" : true,
      "white" : true,
      "cursor-position" : true,
      "word-end" : true,
      "sentence-boundary" : true,
      "sentence-end" : true,
      "word-boundary" : true
    }
  ],
  "output" : {
    "wrapped" : false,
    "ellipsized" : false,
    "hypenated" : false,
    "unknown-glyphs" : 0,
    "width" : 196608,
    "height" : 52428,
    "lines" : [
      {
        "position" : [
          0,
          42947
        ],
        "line" : {
          "start-index" : 0,
          "length" : 9,
          "start-offset" : 0,
          "n-chars" : 9,
          "wrapped" : false,
          "ellipsized" : false,
          "hyphenated" : false,
          "justified" : false,
          "paragraph-start" : true,
          "paragraph-end" : true,
          "direction" : "ltr",
          "runs" : [
            {
              "offset" : 0,
              "length" : 4,
              "text" : "Brat",
              "bidi-level" : 0,
              "gravity" : "south",
              "language" : "en-us",
              "script" : "Latn",
              "font" : {
                "description" : "Cantarell Bold 32",
                "checksum" : "5bcb6ee14ee9d210b2e91d643de1fe456e9d1aea770983fdb05951545efebbe2",
                "variations" : {
                  "wght" : 8374
                }
              },
              "flags" : 0,
              "y-offset" : 0,
              "start-x-offset" : 0,
              "end-x-offset" : 0,
              "glyphs" : [
                {
                  "glyph" : 29,
                  "width" : 28672,
                  "is-cluster-start" : true,
                  "log-cluster" : 0
                },
                {
                  "glyph" : 409,
                  "width" : 17408,
                  "is-cluster-start" : true,
                  "log-cluster" : 1
                },
                {
                  "glyph" : 244,
                  "width" : 23552,
                  "is-cluster-start" : true,
                  "log-cluster" : 2
                },
                {
                  "glyph" : 430,
                  "width" : 16384,
                  "is-cluster-start" : true,
                  "log-cluster" : 3
                }
              ]
            },
            {
              "offset" : 4,
              "length" : 5,
              "text" : "wurst",
              "bidi-level" : 0,
              "gravity" : "south",
              "language" : "en-us",
              "script" : "Latn",
              "font" : {
                "description" : "Cantarell Bold Italic 32",
                "checksum" : "5bcb6ee14ee9d210b2e91d643de1fe456e9d1aea770983fdb05951545efebbe2",
                "matrix" : [
                  1,
                  0.20000000000000001,
                  0,
                  1,
                  0,
                  0
                ]
              },
              "flags" : 0,
              "y-offset" : 0,
              "start-x-offset" : 0,
              "end-x-offset" : 0,
              "glyphs" : [
                {
                  "glyph" : 467,
                  "width" : 33792,
                  "is-cluster-start" : true,
                  "log-cluster" : 0
                },
                {
                  "glyph" : 438,
                  "width" : 24576,
                  "is-cluster-start" : true,
                  "log-cluster" : 1
                },
                {
                  "glyph" : 409,
                  "width" : 16384,
                  "is-cluster-start" : true,
                  "log-cluster" : 2
                },
                {
                  "glyph" : 417,
                  "width" : 20480,
                  "is-cluster-start" : true,
                  "log-cluster" : 3
                },
                {
                  "glyph" : 430,
                  "width" : 15360,
                  "is-cluster-start" : true,
                  "log-cluster" : 4
                }
              ]
            }
          ]
        }
      }
    ]
  }
}
