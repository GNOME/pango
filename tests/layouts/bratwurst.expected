Bratwurst
--- parameters

wrapped: 0
ellipsized: 0
lines: 1
width: 102400

--- attributes

range 0 4
range 4 2147483647
4 4294967295 style italic
range 2147483647 2147483647

--- directions

0 0 0 0 0 0 0 0 0 

--- cursor positions

0(0) 1(0) 2(0) 3(0) 4(0) 5(0) 6(0) 7(0) 8(0) 8(1) 

--- lines

i=1, index=0, paragraph-start=1, dir=ltr 'Bratwurst'

--- runs

i=1, index=0, chars=4, level=0, gravity=south, flags=0, font=OMITTED, script=latin, language=en-us, 'Brat'
i=2, index=4, chars=5, level=0, gravity=south, flags=0, font=OMITTED, script=latin, language=en-us, 'wurst'
i=3, index=9, no run, line end
